front/src/modules/timesStops/TimesStopsOutput.tsx

```diff
@@ -31,17 +35,38 @@ const TimesStopsOutput = ({
   simulatedPathItemTimes,
   operationalPointsOnPath,
 }: TimesStopsOutputProps) => {
-  const rows = useOutputTableData(
+  const useNewTimesStopsTable = useSelector(getUseNewTimesStopsTable);
+
+  // Only call the hook that corresponds to the active table to avoid unnecessary computation
+  const legacyRows = useOutputTableData({
     infraId,
     isValid,
     selectedTrain,
     simulatedTrain,
     simulatedPathItemTimes,
-    operationalPointsOnPath
-  );
+    operationalPointsOnPath,
+    skip: useNewTimesStopsTable,
+  });
+
+  const newRows = useTimesStopsTableData({
+    infraId,
+    isValid,
+    selectedTrain,
+    simulatedTrain,
+    simulatedPathItemTimes,
+    operationalPointsOnPath,
+    skip: !useNewTimesStopsTable,
+  });
+
+  if (useNewTimesStopsTable) {
+    return (
+      <TimesStopsTable rows={newRows} dataIsLoading={!selectedTrain || newRows.length === 0} />
```

Comment:
`selectedTrain` is always defined I think
```suggestion
      <TimesStopsTable rows={newRows} dataIsLoading={newRows.length === 0} />
```

Reply:
fixed